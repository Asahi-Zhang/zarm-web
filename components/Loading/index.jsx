
import React, { Component, PropTypes } from 'react';
import Mask from '../Mask';

class Loading extends Component {

  render () {
    const { width, height, message } = this.props;
    const bodyStyle = {
      'position'   : 'absolute',
      'left'       : '50%',
      'top'        : '50%',
      'marginLeft' : - width / 2,
      'marginTop'  : - height / 2,
      'width'      : width,
      'height'     : height,
    };

    return (
      <div className="ui-loading">
        <div className="ui-loading-body" style={bodyStyle}>
          <br />
          {message}
        </div>
        <Mask type="light" />
      </div>
    );
  }

}

Loading.propTypes = {
  message : PropTypes.string,
  width   : PropTypes.number,
  height  : PropTypes.number,
};

Loading.defaultProps = {
  message : '加载中',
  width   : 80,
  height  : 80,
};

export default Loading;

