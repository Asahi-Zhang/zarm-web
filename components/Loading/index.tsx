import React, { Component } from 'react';
import PropsType from './PropsType';

class Loading extends Component<PropsType, any> {
  static defaultProps = {
    visible: false,
  };

  render() {
    const { visible, children, style } = this.props;

    return visible ? (
      <div className="ui-loading" style={style}>
        <div className="ui-loading-spins">
          <span className="ui-loading-spin ui-loading-spin-first" />
          <span className="ui-loading-spin ui-loading-spin-second" />
          <span className="ui-loading-spin ui-loading-spin-third" />
        </div>
        <div className="ui-loading-inner">{children}</div>
      </div>
    ) : (
      children
    );
  }
}

export default Loading;
