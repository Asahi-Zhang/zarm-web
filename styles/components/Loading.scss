
.ui-loading {
  position: relative;
  
  .ui-loading-spins {
    position: absolute;
    top: 50%;
    height: 20px;
    line-height: 20px;
    margin-top: -10px;
    z-index: 4;
    text-align: center;
    width: 100%;
    opacity: 1;
    visibility: visible;
    user-select: none;
    
    .ui-loading-spin {
      animation: uiLoadingSpinPulse 1.2s infinite cubic-bezier(.78, .14, .15, .86);
      width: 8px;
      height: 8px;
      border-radius: 50%;
      display: inline-block;
      background-color: #2db7f5;

      &.ui-loading-spin-second {
        animation-delay: 200ms;
      }

      &.ui-loading-spin-third {
        animation-delay: 400ms;
      }
    }

    .ui-loading-spin + .ui-loading-spin {
      margin-left: 8px;
    }
  }

  .ui-loading-inner {
    transition: all .3s cubic-bezier(.78, .14, .15, .86);
    opacity: .7;
    -webkit-filter: blur(1px);
    filter: blur(1px);
    filter: progid\:DXImageTransform\.Microsoft\.Blur(PixelRadius\=1, MakeShadow\=false);
    pointer-events: none;
    user-select: none;
  }
}

@keyframes uiLoadingSpinPulse {
  0%, 80%, 100% { opacity: 0 }
  40% { opacity: 1 }
}