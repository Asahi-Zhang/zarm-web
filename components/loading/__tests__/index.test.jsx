import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import Loading from '../index';

describe('Loading', () => {
  it('renders hidden Loading correctly', () => {
    const wrapper = render(
      <Loading>
        这里是内容
      </Loading>,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Loading of different size correctly', () => {
    const wrapper = render(
      <Loading size='lg' visible>
        这里是内容
      </Loading>,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
