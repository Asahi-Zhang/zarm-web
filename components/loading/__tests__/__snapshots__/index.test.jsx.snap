// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading renders Loading of different size correctly 1`] = `
<div
  class="zw-loading zw-loading--lg"
>
  <div>
    <div
      class="zw-loading__mask"
    >
      <div
        class="zw-loading__spin"
      >
        <span
          class="zw-loading__svg zw-activity-indicator zw-activity-indicator--lg zw-activity-indicator__circular--loading"
        >
          <svg
            class="zw-activity-indicator__circular--lg"
            viewBox="31 31 62 62"
          >
            <circle
              cx="62"
              cy="62"
              fill="none"
              r="28.5"
              style="stroke-width:5"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
  <div
    class="zw-loading__container zw-loading--blur"
  >
    这里是内容
  </div>
</div>
`;

exports[`Loading renders hidden Loading correctly 1`] = `
<div
  class="zw-loading zw-loading--md"
>
  <div
    class="zw-loading__container"
  >
    这里是内容
  </div>
</div>
`;
